### PR TITLE
[payload_forward] forward payload messages (decode when possible) from longrange/satcomm modem to external APP

### PR DESCRIPTION
**Pull request by dewagter**

### CODE REVIEW

**Comment by gautierhattenberger** [comment]
"payload" decoding seems to be limited to displaying an image thumbnail. Do you have other plans for this tool ? If not, name and documentation should be more specific. What about making a message dedicated to images ?

**Comment by flixr** [comment]
Seems to depend on a new `PAYLOAD` message in pprzlink...

**Comment by dewagter** [comment]
also depends on: https://github.com/paparazzi/pprzlink/pull/34

File: sw/ground_segment/python/payload_decoder/payload_decoder.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+UDP_IP = "10.0.0.3"
+UDP_PORT = 32000
+
+print "UDP target IP:", UDP_IP
+print "UDP target port:", UDP_PORT
+
+sock = socket.socket(socket.AF_INET, # Internet
+             socket.SOCK_DGRAM) # UDP
+
+sys.path.append(os.getenv("PAPARAZZI_HOME") + "/sw/ext/pprzlink/lib/v1.0/python")
```
Comment (gautierhattenberger) [inline sw/ground_segment/python/payload_decoder/payload_decoder.py]: should be PAPARZZI_SRC

File: sw/ground_segment/python/payload_decoder/payload_decoder.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+        self.thumb = ThumbNailFromPayload()
+        self.data = [];
+
+        self.w = WIDTH
+        self.h = WIDTH
+
+        wx.Frame.__init__(self, id=-1, parent=None, name=u'PayloadDecoder',
+                          size=wx.Size(self.w, self.h), title=u'Payload Decoder')
+
+        ico = wx.Icon(os.getenv("PAPARAZZI_HOME") + "/sw/ground_segment/python/payload_decoder/camera.ico", wx.BITMAP_TYPE_ICO)
```
Comment (gautierhattenberger) [inline sw/ground_segment/python/payload_decoder/payload_decoder.py]: should be PAPARAZZI_SRC

File: sw/ground_segment/python/payload_decoder/payload_decoder.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+    def __init__(self):
+
+        self.thumb = ThumbNailFromPayload()
+        self.data = [];
+
+        self.w = WIDTH
+        self.h = WIDTH
+
+        wx.Frame.__init__(self, id=-1, parent=None, name=u'PayloadDecoder',
```
Comment (gautierhattenberger) [inline sw/ground_segment/python/payload_decoder/payload_decoder.py]: As mentioned before, maybe this tool should be called "Thumbnails decoder" or equivalent, but something closer to the actual purpose of the program.

File: sw/ground_segment/python/payload_decoder/payload_decoder.py (hunk 2 above)
Comment (flixr) [inline sw/ground_segment/python/payload_decoder/payload_decoder.py]: here it can be just relative to the dir of this file...

**Comment by dewagter** [comment]
- [x] README
 - [x] rebase
 - [x] pep8 indent and 'is not None'

anything else?

**Pull request merged**